inginious/frontend/user_manager.py

```diff
@@ -451,6 +489,56 @@ def bind_user(self, auth_id, user):
 
         return True
 
+    def revoke_binding(self, username, binding_id):
+        """
+        Revoke a binding method for a user
+        :param binding_id: The binding method id
+        :param username: username of the user
+        :return: Boolean if error occurred and message if necessary
+        """
+        user_data = self._database.users.find_one({"username": username})
+        if binding_id not in self.get_auth_methods().keys():
+            error = True
+            msg = _("Incorrect authentication binding.")
+        elif user_data is not None and (len(user_data.get("bindings", {}).keys()) > 1 or "password" in user_data):
+            user_data = self._database.users.find_one_and_update(
+                {"username": username},
+                {"$unset": {"bindings." + binding_id: 1}},
+                return_document=ReturnDocument.AFTER)
+            msg = ""
+            error = False
+        else:
+            error = True
+            msg = _("You must set a password before removing all bindings.")
+        return error, msg
+
+    def delete_user(self, username):
```

Comment:
This removes the user from the ``users`` collection but does not removes any other trace from the database, and would allow any new user to get an old submission history by selecting the right username at registration time.

The appropriate piece of code already exists and can be fetched from : https://github.com/UCL-INGI/INGInious/blob/master/inginious/frontend/pages/preferences/delete.py#L17